NCGenerics: more test fixes (v2)

kavon opened the pull request:
base branch is https://github.com/apple/swift/pull/71371

kavon commented:
@swift-ci smoke test

kavon committed:
NCGenerics: handle @_moveOnly correctly

We let TypeCheckAttr diagnose illega uses of the attribute. In addition,
we need to exclude protocols from inferring a marking, because when we
build the StructuralRequirements for a protocol, we ignore the
"marking" this we'll miss this legacy attribute.

fixes Sema/moveonly_decl_attr.swift

slavapestov commented on lib/Sema/TypeCheckDecl.cpp:
```diff
@@ -928,9 +928,8 @@ InvertibleAnnotationRequest::evaluate(Evaluator &evaluator,
-      assert((isa<StructDecl, EnumDecl, ClassDecl>(decl)));
-
+    if (auto attr = decl->getAttrs().getAttribute<MoveOnlyAttr>();
+        attr && isa<StructDecl, EnumDecl, ClassDecl>(decl)) {
```
This will be clearer with a nested `if` I think.

kavon committed:
NCGenerics: omit inverses on invertible protocols

The interface files were still printing inverses on the invertible
protocols, when that's not required. Doing so yielded warnings during
the build because `~Escapable` isn't yet ready for appearing in the
stdlib.

kavon committed:
NCGenerics: avoid UGT's in queries

An unbound generic type is not something that should end up in
`isNoncopyable` or `isEscapable` queries because there are no
conformances for such a type; it's unbound!

fixes test/Constraints/closures.swift

kavon commented:
@swift-ci smoke test

kavon commented:
@swift-ci smoke test

kavon committed:
NCGenerics: reenable round-trip for mangling

kavon commented:
@swift-ci smoke test

kavon commented:
preset=ncgenerics,smoketest=macosx
@swift-ci Please test with preset macOS Platform

kavon commented:
@swift-ci smoke test
